### ISSUE TITLE
Open RTAMO flow to all listed add-ons

### ISSUE THREAD

**Issue by diox** [mozilla/addons-server#17826]
RTAMO allows an user installing Firefox to be prompted to install the add-on they installed Firefox from (on Windows). This is limited to badged add-ons at the moment, and we want to extend it to all listed.

This needs to be guarded by a separate waffle switch so that the feature can be turned off. The switch needs to be different from the one guarding the RTAMO feature itself, so that RTAMO can be turned off independently. The switch needs to be `off` by default, we'll turn it on when ready.

**Comment by diox** [comment]
Note: Waffle switch is `return-to-amo-for-all-listed`

**Comment by ioanarusiczki** [comment]
@diox 

I enabled on -dev earlier today `return-to-amo-for-all-listed` using `admin/models/waffle/switch/`

I could see in about:welcome page the recommended, by Firefox, verified addons (as before, the flow seems to be working with the labeled extensions, works with a recommended theme too) but when I tried with extensions without a label I did not get the same results (I tried 2 extensions, a theme and a dict)

I applied the same steps as described into https://bugzilla.mozilla.org/show_bug.cgi?id=1724169, steps also described [here, in detail](https://mana.mozilla.org/wiki/pages/viewpage.action?pageId=150537195) but I'll try to make it shorter:

1. opened sandbox then with Edge I installed the [extension](https://bugzilla.mozilla.org/show_bug.cgi?id=1724169#c15) and followed Will's instructions (including clearing all cookies for mozilla.org)
2. I went to -dev and opened a detail page clicked the blue button 
3. installed FF (release), then closed it and downloaded FF Nightly 
4. with FF Nightly from about:config I added the pref `extensions.getAddons.get.url` for `services.addons-dev.allizom` 
5. in a new tab I opened about:welcome page

**Comment by diox** [comment]
Mmm, it could have been caching if you tried first with the waffle off for that add-on ? API responses (that RTAMO depends on) are cached for 3 minutes, and are updated in the background, so the first response back after the caching duration could be stale as well.

Could you capture the network traffic that goes to services.addons-dev.allizom.org when you're testing this ?

The API seems to behave like expected, with the waffle enabled `curl -sg 'https://services.addons-dev.allizom.org/api/v5/addons/search/?guid=rta:ezA0YjMzMDI4LThhMmQtNGU3NC1iYjAxLWIzZjE5YmI3NDM1OX0'` returns data about add-on with guid `{04b33028-8a2d-4e74-bb01-b3f19bb74359}` which is not promoted.

**Comment by ioanarusiczki** [comment]
@diox 
I just realized I left both switched active in admin. Now that I disabled `return-to-amo` and left `return-to-amo-for-all-listed` enabled I got the flow for an extension not labeled. Seems to be working for labeled too (at least for a recommended one) 
So I'll try to cover more addon scenarios tomorrow just to be sure. 

Seems like with both switched enabled - only the labeled one were working.

**Comment by diox** [comment]
To be clear: 
- If `return-to-amo` is `off`, RTAMO is completely disabled, the other switch doesn't matter.
- If `return-to-amo` is `on`, but `return-to-amo-for-all-listed` is `off`: RTAMO is enabled only for promoted
- If `return-to-amo` is `on`, but `return-to-amo-for-all-listed` is `on`: RTAMO is enabled for all listed

Caching may mess with this: if the same add-on has been previously tested, whatever result you had before _may_ persist for the first following request even if you change the waffle switches. To make it easier to test I recommend using different add-ons every time.

**Comment by ioanarusiczki** [comment]
@diox 

I tested this again today:

- with both switched off, RTAMO is disabled. I verified it for a recommended addon on -dev and another one on -stage. Same results: the addon is not available in about:welcome page.
- with `return-to-amo` on and `return-to-amo-for-all-listed`  off : I could see that recommended addons are available in about:welcome but extensions without a label are not.
- both switched on: promoted addons are available. Extensions, themes, dictionaries and lang packs without being promoted - are displayed. 
- with `return-to-amo` off and `return-to-amo-for-all-listed`  on.  The feature is not working. I tested for a recommended then a verified extension and a lang pack which was not promoted.  


I verified this with sandbox (using Edge and FF Nightly on AMO dev and AMO stage) - reopened a new sandbox session for each addon tested. I always picked a different addon too. 

I don't know what happened yesterday, but this time it looks like all scenarios are working as expected. 

I left both switches enabled.
